src/model/entities/part.ts

```diff
@@ -302,13 +317,30 @@ export class Part {
     this.endState = endState
   }
 
+  public getInTransition(): InTransition {
+    return this.inTransition
+  }
+
   public reset(): void {
+    if (!this.defaultPart) {
+      return
+    }
+
     this.executedAt = 0
     this.playedDuration = 0
+    this.inTransition = this.defaultPart.inTransition
+    this.timings = this.defaultPart.timings
+    this.endState = undefined
+
     this.pieces = [
       ...this.pieces.filter(piece => piece.isPlanned),
       ...this.replacedPlannedPieces
-    ]
+    ].filter(piece => this.defaultPart!.ingestedPieces.some(ingestPiece => ingestPiece.id === piece.id))
+      .map(piece => {
+        const ingestedPiece: IngestedPiece = this.defaultPart!.ingestedPieces.find(ingestPiece => ingestPiece.id === piece.id)!
+        piece.resetFromIngestedPiece(ingestedPiece)
+        return piece
+      })
```

Comment:
I think this has a size now that i would benefit from its own aptly named method. The assignment to one attribute has grow to 10 lines of code. At a minimum, the function of the map, should be moved out. As long as that is done, i am happy.

Reply:
I have extracted it to a method